app/controllers/concerns/listings_toolkit.rb

```diff
@@ -124,7 +121,7 @@ def first_publish(cost)
     author = @listing.author
     available_user_credits = author.is_a?(Organization) ? current_user.credits.unspent.size : 0
 
-    if author.credits.unspent.size >= cost
+    if author.enough_credits?(cost)
```

Comment:
Reusing the existing `User#enough_credits?` method. I'd prefer eventually completely removing this from `User`.